### PR TITLE
Add S3 and Cloudfront Subchecks

### LINKED ISSUE #69: Cloudfront und S3
Hi,

wir benötigen die Möglichkeit AWS Cloudfront und S3 zu überwachen. Bitte um entsprechende Erweiterung.

Gruß,

Gunter Geib, DATEV

### PR DESCRIPTION
**Pull request by pdorschner**
Implements new subcommands to check:

 * S3 Buckets
 * Cloudfront

See Iusse [Cloudfront und S3 #69](https://github.com/NETWAYS/check_cloud_aws/issues/69)

Fixes #69

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,14 +68,115 @@ Type: t2.micro
 AutoScaling: (none)
 ```
 
+### S3
+
+In the bucket context, calculates the objects inside the bucket and alerts if the size reaches the threshold.
```
Comment (martialblog) [inline README.md:73]: calculates the size of the objects, right?

File: README.md [hunk 2]
```diff
@@ -66,14 +68,115 @@ Type: t2.micro
...
+In the object context, each object inside a bucket will be checked if the size reaches the threshold.
```
Comment (martialblog) [inline README.md:74]: ...will be checked if **its** size reaches...

File: README.md [hunk 3]
```diff
@@ -66,14 +68,115 @@ Type: t2.micro
...
+
+````
+$ check_cloud_aws s3 object --perfdata --prefix 'test' -b 'my-aws-testbucket1'
+OK - Found 3 objects - critical 0 - warning 0 | test-file2.jpg=20MB;800;1024 test-file3.gif=10MB;800;1024 test-file5.rtf=10MB;800;1024
+
+````
+
+### Cloudfront
+
+Checks a specific or multiple cloudfront distribution from a region. When the states is `disabled`, or is `InProgress`,
```
Comment (martialblog) [inline README.md:136]: distribution**s**, same as in the help text

**Review by martialblog — CHANGES_REQUESTED** [review]
:+1:  Just some minor stuff

**Comment by martialblog** [comment]
@pdorschner could you also please rebase/squash the Commit bbc16db839b23e9

File: README.md (hunk 1 above)
Comment (pdorschner) [reply README.md:73]: Yes, each object inside the bucket will be [summarized](https://github.com/NETWAYS/check_cloud_aws/blob/feature/cloudfront/internal/s3/bucket.go#L29)

Comment (martialblog) [reply README.md:73]: should be made clear in the description

**Review by martialblog — APPROVED** [review]
:+1: Squash it at merge

**Pull request merged**